src/init-action.ts

```
@@ -588,32 +587,25 @@ async function run() {
       core.exportVariable(bmnVar, value);
     }
 
-    // For rust: set CODEQL_ENABLE_EXPERIMENTAL_FEATURES, unless codeql already supports rust without it
+    // For rust, if running a version prior to public preview (2.22.1) set CODEQL_ENABLE_EXPERIMENTAL_FEATURES
```

I am not sure that this comment accurately describes what is currently implemented. As far as I can tell you check whether the actual CLI version is less than `minVer` and if so throw a `ConfigurationError`. That means we never get to the `CODEQL_ENABLE_EXPERIMENTAL_FEATURES` check. 

If that is the case, and that behaviour is unintentional, then you should obviously fix it, but I'd also like to perhaps see this refactored with a test to make sure it does the right thing.

---

there are 2 versions involved:
* prior to 2.19.3, there is no rust support whatsoever, which means that if we are requesting rust we indeed need to throw a configuration error
* between 2.19.3 and 2.22.1, rust support is experimental, which means it needs to be set explicitly. What I now see, is that if `!(await codeql.resolveLanguages()).rust`, that actually entails already that `CODEQL_ENABLE_EXPERIMENTAL_FEATURES` is not set (otherwise `codeql resolve languages` would provide an entry for `rust`, which makes the `process.env[envVar] !== "true"` check redundant.
* after 2.22.1, `(await codeql.resolveLanguages()).rust` will always be set

If we forego the `"Experimental Rust analysis enabled"` log line entirely (there's probably no burning need for it), we can simplify by getting rid of the second if. And the comment needs a rewrite to be clearer